addons/calendar_sms/models/calendar.py

```diff
@@ -17,10 +17,11 @@ def _sms_get_default_partners(self):
         """
         return self.mapped('attendee_ids').filtered(lambda att: att.state != 'declined').mapped('partner_id')
 
-    def _do_sms_reminder(self):
+    def _do_sms_reminder(self, alert):
         """ Send an SMS text reminder to attendees that haven't declined the event """
         for event in self:
             event._message_sms_with_template(
+                template=self.env['calendar.alarm'].browse(alert['alarm_id']).sms_template_id,
```

Comment:
if template is set, template_xmlid will not be used, unless self.env['calendar.alarm'].browse(alert['alarm_id']).sms_template_id return False.
So if it's possible to return False, you can keep it like this, otherwise, you can remove template_xmlid from the arguments

Reply:
done, I made the template fields required in the view and removed "tempalte_xmlid"